test/io.test.js

```diff
@@ -161,6 +161,18 @@ describe('IO', function(){
                 block.write(new Buffer([0x94, 0x02]));
                 decoder.readLong().should.equal(138);
             })
+
+            // http://lucene.apache.org/core/3_5_0/fileformats.html#VInt
+            it('should decode and return a positive long using variable-length + zigzag', function () {
```

Comment:
You'll have to explain zigzag encoding to me - but I thought the idea was that you encode numbers as unsigned integers - where is this happening? I'm assuming `0x80` vs `0x81` is where the sign is, or is this internal in the `readLong`?